gcc7 util.c - ‘%s’ directive writing up to 255 bytes into a region of size 232

**Environment**

Fedora 26 x86_64
gcc 7.1.1
git hash faef420b1e4eecee4e2718083ee5bc11d22e9c1e

**Error**

```
gcc -DHAVE_CONFIG_H -I. -I..    -Wall -Werror -DVERSIONCOMMIT=\""0.8.1, commit: `git describe --dirty --always --tags 2> /dev/null || true`"\" -g -O2 -MT util.o -MD -MP -MF $depbase.Tpo -c -o util.o util.c &&\
mv -f $depbase.Tpo $depbase.Po
util.c: In function ‘online_cpu’:
util.c:514:42: error: ‘%s’ directive writing up to 255 bytes into a region of size 232 [-Werror=format-overflow=]
   sprintf(path, "/sys/devices/system/cpu/%s/online", entry->d_name);
                                          ^~
util.c:514:3: note: ‘sprintf’ output between 32 and 287 bytes into a destination of size 256
   sprintf(path, "/sys/devices/system/cpu/%s/online", entry->d_name);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
util.c: In function ‘online_memory’:
util.c:550:45: error: ‘%s’ directive writing up to 255 bytes into a region of size 229 [-Werror=format-overflow=]
   sprintf(path, "/sys/devices/system/memory/%s/online", entry->d_name);
                                             ^~
util.c:550:3: note: ‘sprintf’ output between 35 and 290 bytes into a destination of size 256
   sprintf(path, "/sys/devices/system/memory/%s/online", entry->d_name);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/util.c
+++ b/src/util.c
@@ -511,7 +511,7 @@ void online_cpu(void)
 		ret = sscanf(entry->d_name, "cpu%d", &num);
 		if (ret < 1 || num == 0) /* skip none cpu%d and cpu0 */
 			continue;
-		sprintf(path, "/sys/devices/system/cpu/%s/online", entry->d_name);
+		sprintf(path, "/sys/devices/system/cpu/cpu%d/online", num);
 		fd = open(path, O_RDWR);
 		if (fd < 0) {
 			fprintf(stderr, "open %s failed\n", path);
@@ -547,7 +547,7 @@ void online_memory(void)
 		ret = sscanf(entry->d_name, "memory%d", &num);
 		if (ret < 1 || num == 0) /* skip none memory%d and memory0 */
 			continue;
-		sprintf(path, "/sys/devices/system/memory/%s/online", entry->d_name);
+		sprintf(path, "/sys/devices/system/memory/memory%d/online", num);
 		fd = open(path, O_RDWR);
 		if (fd < 0) {
 			fprintf(stderr, "open %s failed\n", path);

